presto-openapi/src/main/resources/query.yaml

```diff
@@ -0,0 +1,120 @@
+          required: true
+          schema:
+            type: string
+          description: The ID of the query to cancel
+      responses:
+        '204':
+          description: Query canceled successfully
+    get:
+      summary: Retrieve information about a specific query
+      description: Retrieves detailed information about the query identified by queryId
+      parameters:
+        - name: queryId
+          in: path
+          required: true
+          schema:
+            type: string
+          description: The ID of the query to retrieve
+      responses:
+        '200':
+          description: Detailed query information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/QueryInfo'
+        '404':
+          description: Query not found
+  /v1/query/{queryId}/killed:
+    put:
+      summary: Kill a specific query
+      description: Kills the query identified by queryId with a provided message
+      parameters:
+        - name: queryId
+          in: path
+          required: true
+          schema:
+            type: string
+          description: The ID of the query to kill
+      requestBody:
+        required: true
+        content:
+          text/plain:
+            schema:
+              type: string
+              description: Message explaining why the query is being killed
+      responses:
+        '200':
+          description: Query killed successfully
+  /v1/query:
+    get:
+      summary: Returns a list of query resource information
+      description: Retrieves information about all queries with the ability to filter and limit the result list using optional parameters
+      parameters:
+        - name: state
+          in: query
+          required: false
+          schema:
+            type: string
+          description: Filter queries by state
+        - name: limit
+          in: query
+          required: false
+          schema:
+            type: integer
+          description: Limit the number of queries returned
+      responses:
+        '200':
+          description: A list of queries
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/BasicQueryInfo'
+
+components:
+  schemas:
+    BasicQueryInfo:
+      type: object
+      properties:
+        state:
+          type: string
+          description: The state of the query
+        queryStats:
+          $ref: '#/components/schemas/QueryStats'
+          description: Statistics about the query
+    QueryInfo:
+      type: object
+      properties:
+        queryId:
+          type: string
+          description: The ID of the query
+        state:
+          type: string
+          description: The state of the query
+        queryStats:
+          $ref: '#/components/schemas/QueryStats'
+          description: Statistics about the query
+    QueryStats:
```

Comment:
I think this should be `BasicQueryStats`. There's also a lot of missing fields. All the constructor parameters annotated with `@JsonProperty` should be included in this schema